src/settings.rs

```diff
@@ -105,9 +109,47 @@ impl kubewarden::settings::Validatable for Settings {
             return Err("Signatures must not be empty".to_string());
         }
 
-        // TODO: when a certificate is being used, ensure the certificate
-        // can be trusted. This requires a new waPC function being written
-
+        // when a certificate is being used, ensure it can be trusted
+        for signature in self.signatures.iter() {
+            if let Signature::Certificate(s) = signature {
+                // build sdk structs:
+                let cert = SDKCert {
+                    encoding: CertificateEncoding::Pem,
+                    data: s.certificate.clone().into_bytes(),
+                };
+                let cert_chain_opt: Option<Vec<SDKCert>> = match &s.certificate_chain {
+                    Some(chain_vec) => {
+                        // build vec of sdk certs:
+                        let mut chain_sdk: Vec<SDKCert> = vec![];
+                        chain_vec.iter().for_each(|d| {
+                            chain_sdk.push(SDKCert {
+                                encoding: CertificateEncoding::Pem,
+                                data: d.clone().into_bytes(),
+                            });
+                        });
+                        Some(chain_sdk)
+                    }
+                    None => None,
+                };
+                match verify_cert(cert, cert_chain_opt, Some(Utc::now().to_rfc3339())) {
```

Comment:
We can't verify with the `not_after` set to `now`. That's because a signature could have been produced at the right time (when the certificate was valid), but now the same certificate could be expired.

We have to give `None` here. The actual certificate verification with the `not_after` will be done later, during validate, using the rekor data provided by the signed artifact

Reply:
true! done https://github.com/kubewarden/verify-image-signatures/pull/47/commits/f0867ff9e0fb44ab3819b32e139970da1f367fcb.